utility/data_fetcher.go

```diff
@@ -0,0 +1,167 @@
+package utility
+
+import (
+	"errors"
+	"time"
+)
+
+var (
+	typeErr = errors.New("Invalid type for field.")
+	nilErr  = errors.New("Mandatory field missing.")
+)
+
+type DataFetcher struct {
+	Err error
+}
+
+func (d *DataFetcher) Float64(base map[string]interface{}, key string, keys ...string) float64 {
+	val := d.getDeep(base, keys...)[key]
+	if valFloat, ok := val.(float64); ok {
+		return valFloat
+	} else if val == nil {
```

Comment:
I think this case is not needed, neither `nilErr` error (attribute required-ness already happens in json schema validation)

Reply:
You're right that this should be caught in the validation already. However, this method doesn't know anything about a validation, and I'd rather keep it this way, to still have a defined handling in case the validation changes or a bug gets introduced.

Reply:
The goal of validation is not that you can make assumptions about the data later on? 

We are making assumptions already, explicitly or implicitly (eg: we assume we can save this data in ES conforming its mapping because we have validated it). So this double-checking is actually redundant.

Reply:
Not checking for a nil or a type error could lead to storing incomplete data in ES as we would go on with the processing, although some value that should be given might be missing.

I'd rather keep this check in than taking the risk of having unexpected data in ES.
The validation implementation is completely separated from this implementation, and as long as we do not get the gostructs as a result out of the validation, I'm afraid we have to do the same validations twice, or risking that the logic can drift apart. 

Regarding assumptions about being able to store the data in ES - we do have integration tests, that ensure the data can actually be stored, so I don't really get your point here.